packages/shared/src/graphql/posts.ts

```diff
@@ -91,6 +99,27 @@ export const POST_BY_ID_QUERY = gql`
   }
 `;
 
+export const POST_UPVOTES_BY_ID_QUERY = gql`
+  query postUpvotes($id: String!, $after: String, $first: Int) {
+    postUpvotes(id: $id, after: $after, first: $first) {
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+      edges {
+        node {
+          user {
+            name
+            username
+            bio
+            image
+          }
+        }
+      }
+    }
```

Comment:
We can reuse this structure with something called `Fragment`. I use it in the codebase so check for it